lib/raml/parser.rb

```diff
@@ -11,7 +11,14 @@ class Parser
 
     def initialize
       Psych.add_domain_type 'include', 'include' do |_, value|
-        File.read(value)
+        content = File.read(value)
+
+        case File.extname(value)
+        when *%w[.yaml .yml .raml]
```

Comment:
It seems like `.yaml` and `.yml` should be passed through `YAML.load_file`, but `.raml` should probably be parsed as raml (using `Raml::Parser.parse_file`) - would you agree?